enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,597 @@
+  both cost and performance. Customers can use devices on the market from OEMs
+  like Dell that supply an all in one unit with 2 compute and 1 lower powered
+  compute for this deployment scenario.
+
+### Non-Goals
+
+The below goals are not intended to be worked on now, but might be expansion
+ideas for future features.
+
+- Running the arbiter node offsite.
+- Running a virtualized arbiter node on the same cluster.
+- Having a single arbiter supporting multiple clusters.
+- Moving from 2 + 1 to a conventional 3 node cluster
+
+## Proposal
+
+The main focus of the enhancement is to support edge deployments of individual
+OpenShift HA clusters at scale, and to do so in a cost effective way. We are
+proposing doing this through the creation of a new node role type called an
+`node-role.kubernetes.io/arbiter` node as a heterogenous quasi control plane
+configuration. The arbiter will run the critical components that help maintain
+an HA cluster, but other platform pods should not be scheduled on the arbiter
+node. The arbiter node will be tainted to make sure that only deployments that
+tolerate that taint are scheduled on the arbiter.
+
+We think creating this new node role would be the best approach for this
+feature. Having a new node role type allows our bootstrap flow to leverage the
+existing MCO and Installer systems to supply specific Ignition files for a
+smaller footprint device. Existing deployments will essentially be blind to this
+node role unless they need to be made aware of it, but critical workloads that
+allow all taints can still run on the node, thus making it easier for smaller
+arbiter nodes to be deployed with out higher memory and cpu requirements.
+
+Components that we are proposing to change:
+
+| Component                                                  | Change                                                                                                          |
+| ---------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------- |
+| [Infrastructure API](#infrastructure-api)                  | Add `HighlyAvailableArbiter` as a new value for `ControlPlaneTopology` and `InfrastructureTopology`             |
+| [Installer](#installer-changes)                            | Update install config API to Support arbiter machine types                                                      |
+| [MCO](#mco-changes)                                        | Update validation hook to support arbiter role and add bootstrapping configurations needed for arbiter machines |
+| [Kubernetes](#kubernetes-change)                           | Update allowed `well_known_openshift_labels.go` to include new node role as an upstream carry                   |
+| [ETCD Operator](#etcd-operator-change)                     | Update operator to deploy operands on both `master` and `arbiter` node roles                                    |
```

Comment:
If the arbiter must not be the leader, then that's an additional feature that needs to be listed and designed.

Reply:
After some discussion on what we want to happen here. We're okay with the Arbiter node etcd member becoming leader. We're under the impression that etcd will be as fast as it's slowest member, then this might be fine for now. I'll document this as an open question, and plan to revisit this as it's own potential EP, wdyt?

@tjungblu Any concerns with this assumption on my part?

Reply:
there is no functionality that can prevent it becoming a leader. I'm mostly concerned about the performance impact reported here: https://github.com/etcd-io/etcd/issues/14501

There are some upstream k8s PRs in the works that enable grpc "circuit breakers", so this will get resolved at some point in the future